src/utils/headers.ts

```diff
@@ -57,37 +88,62 @@ export function filterRequestHeaders(request: CloudFrontRequest): OutgoingHttpHe
 export function updateResponseHeaders(headers: IncomingHttpHeaders, domain: string): CloudFrontHeaders {
   const resultHeaders: CloudFrontHeaders = {}
 
-  for (const name of ALLOWED_RESPONSE_HEADERS) {
-    const headerValue = headers[name]
-    if (headerValue) {
-      resultHeaders[name] = [
+  for (const [key, value] of Object.entries(headers)) {
+    // Lambda@Edge function can't add read-only headers to response to CloudFront
+    // So, such headers from IngressAPI response are filtered out before return the response to CloudFront
+    if (!isHeaderAllowedForResponse(key)) {
+      continue
+    }
+
+    if (key === COOKIE_HEADER_NAME && value !== undefined && Array.isArray(value)) {
+      resultHeaders[COOKIE_HEADER_NAME] = [
+        {
+          key: COOKIE_HEADER_NAME,
+          value: adjustCookies(value, domain),
+        },
+      ]
+    } else if (key == CACHE_CONTROL_HEADER_NAME && typeof value === 'string') {
+      resultHeaders[CACHE_CONTROL_HEADER_NAME] = [
         {
-          key: name,
-          value: headerValue.toString(),
+          key: CACHE_CONTROL_HEADER_NAME,
+          value: updateCacheControlHeader(value),
+        },
+      ]
+    } else if (value) {
+      resultHeaders[key] = [
+        {
+          key: key,
+          value: value.toString(),
         },
       ]
     }
   }
 
-  if (headers[COOKIE_HEADER_NAME] !== undefined) {
-    resultHeaders[COOKIE_HEADER_NAME] = [
-      {
-        key: COOKIE_HEADER_NAME,
-        value: adjustCookies(headers[COOKIE_HEADER_NAME], domain),
-      },
-    ]
-  }
+  return resultHeaders
+}
 
-  if (headers[CACHE_CONTROL_HEADER_NAME] !== undefined) {
-    resultHeaders[CACHE_CONTROL_HEADER_NAME] = [
-      {
-        key: CACHE_CONTROL_HEADER_NAME,
-        value: updateCacheControlHeader(headers[CACHE_CONTROL_HEADER_NAME]),
-      },
-    ]
+function isHeaderAllowedForRequest(headerName: string) {
+  if (READ_ONLY_REQUEST_HEADERS.has(headerName) || BLACKLISTED_HEADERS.has(headerName)) {
+    return false
   }
+  for (let i = 0; i < BLACKLISTED_HEADERS_PREFIXES.length; i++) {
+    if (headerName.startsWith(BLACKLISTED_HEADERS_PREFIXES[i])) {
```

Comment:
I'm agree with coverage tool, we need a test for `BLACKLISTED_HEADERS_PREFIXES` logic.
It will be great to cover `isHeaderAllowedForRequest` function.